targets/p5.js

```diff
@@ -0,0 +1,188 @@
+import {
+  sculptToGLSL,
+  uniformsToGLSL,
+} from "../generators/sculpt.js";
+import {
+  usePBRHeader,
+  useHemisphereLight,
+  sculptureStarterCode,
+  fragFooter,
+} from "../glsl/glsl-lib.js";
+
+export function toGLSL(source) {
+  if (typeof source === "function") {
+    source = source.toString();
+    source = source.slice(source.indexOf("{") + 1, source.lastIndexOf("}"));
+  } else if (!(typeof source === "string")) {
+    throw "sculptToHydraRenderer requires the source code to be a function, or a string";
+  }
+
+  const vert = `#version 300 es
+precision highp float;
+in vec3 aPosition;
+uniform mat4 uModelViewMatrix;
+uniform mat4 uInverseModelViewMatrix;
+uniform mat4 uProjectionMatrix;
+out vec4 worldPos;
+out vec3 sculptureCenter;
+out vec3 cameraPosition;
+void main() {
+  vec4 mvPosition = uModelViewMatrix * vec4(aPosition, 1.0);
+
+  mat4 modelMatrix = uModelViewMatrix;
+
+  worldPos = vec4(aPosition, 1.);
+  sculptureCenter = vec3(0.);
+  cameraPosition = (uInverseModelViewMatrix * vec4(0., 0., 0., 1.0)).xyz;
+
+  gl_Position = uProjectionMatrix * mvPosition;
+}
+`
+
+  let generatedGLSL = sculptToGLSL(source);
+  let frag = `#version 300 es
+precision highp float;
+uniform mat4 uModelViewMatrix;
+uniform mat4 uProjectionMatrix;
+in vec4 worldPos;
+in vec3 sculptureCenter;
+in vec3 cameraPosition;
+out vec4 pc_fragColor;
+` +
+    usePBRHeader +
+    useHemisphereLight +
+    uniformsToGLSL(generatedGLSL.uniforms) +
+    "const float STEP_SIZE_CONSTANT = " +
+    generatedGLSL.stepSizeConstant +
+    ";\n" +
+    "const int MAX_ITERATIONS = " +
+    generatedGLSL.maxIterations +
+    ";\n" +
+    "#define MAX_REFLECTIONS " +
+    generatedGLSL.maxReflections +
+    "\n" + 
+    sculptureStarterCode.replace(/max_dist = [\d\.]+;/, 'max_dist = 1000.0;') +
+    generatedGLSL.geoGLSL +
+    "\n" +
+    generatedGLSL.colorGLSL +
+    "\n" +
+    fragFooter.replace(
```

Comment:
This is another good idea that we could implement in our other targets to reduce duplication. thanks!